classes/PrestaShopLogger.php

```diff
@@ -157,6 +158,7 @@ public static function addLog($message, $severity = 1, $errorCode = null, $objec
         $log->severity = (int) $severity;
         $log->error_code = (int) $errorCode;
         $log->message = $message;
+        $log->hash = $hash;
```

Comment:
```suggestion
        $log->hash = $hash ?: $log->getHash();
```
If hash becomes the new key to find the message it should always be populated by default, the current implementation will always leave the column empty

I'm not even sure we need to add the extra parameter in the method, because it will neveer be used anywhere in the end

So in the end maybe this would be enough:


```suggestion
        $log->hash = $log->getHash();
```

Reply:
The Hash is used only for log messages where we need to check for duplicates. Most messages do not use this as they allow duplicate entries. So we do not need hash keys for all rows and can save the space by saving null.

Reply:
@jolelievre what do you think?